### DISCUSSION TITLE
Middleware with createStore()?

### DISCUSSION THREAD

**Discussion by noconsulate** [pmndrs/zustand#1956] [Q&A]
I was doing just fine with middleware when I was using the create() zustand method as explained in the docs. But I had to switch to using createStore() in order to initialize with props. I managed to get it working perfectly but I can't seem to use any middleware. Is it possible?

**Comment by dai-shi** [comment]
It should be possible: https://tsplay.dev/wO1bRm

<img width="504" alt="image" src="https://github.com/pmndrs/zustand/assets/490574/0064789c-b1e3-4cbe-a7a5-9651495f9bc9">

**Reply by noconsulate** [reply]
I'm sure this works, but I tried it out and haven't figured it out yet. That's not your problem, obviously, I'm just pointing out it doesn't directly fit in the pattern laid out in the docs so maybe at some point the devs will document how to do it. If I figure it out I'll post the code here for posterity. 

Specifically I don't see where WithDevTools is defined.

**Reply by dai-shi** [reply]
Yeah, Vanilla JS usage is in general not well documented, due to the fact that it's not the major usage.
WithDevTools is an internal type which you don't specify explicitly. The TypeScript guide in the docs would help to understand it deeply.

**Reply by noconsulate** [reply]
I urge the devs to document how to use middleware with the createStore() API. All of the examples seem to use create() but I've found zustand works much better when I use createStore() and createContext(). I don't really know what I'm doing with these functions but it works and so I'd be happiest not to have to dig deep with this stuff just to get middleware to work!

**Reply by noconsulate** [reply]
Thanks I'll check it out.

**Reply by dai-shi** [reply]
> zustand works much better when I use createStore() and createContext().

Right, that's not well documented either, I guess. It's in https://docs.pmnd.rs/zustand/guides/initialize-state-with-props, but the title doesn't tell it well.
@dbritto-dev Any thoughts?

**Comment by noconsulate** [comment]
Ok this looks promising , thanks.
